src/command.rs

```diff
@@ -62,80 +65,100 @@ pub async fn init(kind: PackageType, name: Option<PackageName>) -> eyre::Result<
 }
 
 /// Adds a dependency to this project
-pub async fn add(registry: RegistryUri, dependency: &str) -> eyre::Result<()> {
+pub async fn add(registry: RegistryUri, dependency: &str) -> miette::Result<()> {
     let lower_kebab = |c: char| (c.is_lowercase() && c.is_ascii_alphabetic()) || c == '-';
 
     let (repository, dependency) = dependency
         .trim()
         .split_once('/')
-        .wrap_err("Invalid dependency specification")?;
+        .ok_or_else(|| miette!("locator {dependency} is missing a repository delimiter"))?;
 
     ensure!(
         repository.chars().all(lower_kebab),
-        "Repositories must be in lower kebab case"
+        "repository {repository} is not in kebab case"
     );
 
     let repository = repository.into();
 
     let (package, version) = dependency
         .split_once('@')
-        .wrap_err_with(|| format!("Invalid dependency specification: {dependency}"))?;
+        .ok_or_else(|| miette!("dependency specification is missing version part: {dependency}"))?;
 
     let package = package
         .parse::<PackageName>()
-        .wrap_err_with(|| format!("Invalid package name supplied: {package}"))?;
+        .wrap_err_with(|| format!("invalid package name: {package}"))?;
 
     let version = version
         .parse::<VersionReq>()
-        .wrap_err_with(|| format!("Invalid version requirement supplied: {package}"))?;
+        .into_diagnostic()
+        .wrap_err_with(|| format!("not a valid version requirement: {version}"))?;
 
-    let mut manifest = Manifest::read().await?;
+    let mut manifest = Manifest::read()
+        .await
+        .wrap_err("failed to read manifest file")?;
 
     manifest
         .dependencies
         .push(Dependency::new(registry, repository, package, version));
 
-    manifest.write().await
+    manifest
+        .write()
+        .await
+        .wrap_err("failed to write manifest file")
 }
 
 /// Removes a dependency from this project
-pub async fn remove(package: PackageName) -> eyre::Result<()> {
-    let mut manifest = Manifest::read().await?;
+pub async fn remove(package: PackageName) -> miette::Result<()> {
+    let mut manifest = Manifest::read()
+        .await
+        .wrap_err("failed to read manifest file")?;
 
     let match_idx = manifest
         .dependencies
         .iter()
         .position(|d| d.package == package)
-        .wrap_err(eyre::eyre!(
-            "Unable to remove unknown dependency {package:?}"
-        ))?;
+        .ok_or_else(|| miette!("package {package} not in manifest"))?;
 
     let dependency = manifest.dependencies.remove(match_idx);
 
-    PackageStore::uninstall(&dependency.package).await.ok();
+    // if PackageStore::uninstall(&dependency.package).await.is_err() {
+    //     tracing::warn!("failed to uninstall package {}", dependency.package);
+    // }
 
-    manifest.write().await
+    PackageStore::uninstall(&dependency.package).await.ok(); // temporary due to broken test
```

Comment:
Ah. This is bad, can we fix it?

Reply:
yeah but want to do that in a separate PR as it's outside of the scope of this one

Reply:
Can you create a bug report?

Reply:
https://github.com/helsing-ai/buffrs/issues/115